### PR TITLE
Add YOLO object detection model

### LINKED ISSUE #22: Add yolo v3/4
https://github.com/eriklindernoren/PyTorch-YOLOv3/issues/487
https://github.com/Tianxiaomo/pytorch-YOLOv4

### PR DESCRIPTION
**Pull request by senarvi**
## What does this PR do?

This PR adds the YOLO object detection model. The implementation is based on the YOLOv3 and YOLOv4 Darknet implementations, although it doesn't include all the features of YOLOv4. Detection seems to work with weights that have been trained using the Darknet implementation, so the network architecture should be more or less identical. The network architecture is read from a configuration file in the same format as in the Darknet implementation. It supports loading weights from a Darknet model file too, if you don't want to start training from a randomly initialized model.

Fixes #22 

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning-bolts/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? [not needed for typos/docs]
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning-bolts/blob/master/CHANGELOG.md)?

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pl_bolts/datamodules/vocdetection_datamodule.py
Comment (akihironitta) [inline pl_bolts/datamodules/vocdetection_datamodule.py:157]: I think we don't need `Optional` here if we set `[]` as the default value.
```suggestion
        transforms: List[Callable] = [],
```

**Review by akihironitta — COMMENTED** [review]
We recently fixed some of the tests, so would you mind merging master branch? Also, it seems the change here is causing errors in CI, so would you mind resolving that, too?

File: pl_bolts/datamodules/vocdetection_datamodule.py
Comment (senarvi) [reply pl_bolts/datamodules/vocdetection_datamodule.py]: > We recently fixed some of the tests, so would you mind merging master branch? Also, it seems the change here is causing errors in CI, so would you mind resolving that, too?

I fixed the datamodule, but what do you mean by merging master branch? I believe this is based on the head of the master branch.

Comment (senarvi) [reply pl_bolts/datamodules/vocdetection_datamodule.py]: I'm resolving this conversation but let me know if there's still something I need to do.

**Comment by williamFalcon** [comment]
this is awesome. @Borda @kaushikb11 let's get this merged? and can we add as backbones available in Flash?

File: pl_bolts/models/detection/yolo/yolo_layers.py
Comment (oke-aditya) [inline pl_bolts/models/detection/yolo/yolo_layers.py:58]: hmm, this is same as `torchvision.ops.box_iou` ? I guess ?

File: pl_bolts/models/detection/yolo/yolo_layers.py
Comment (oke-aditya) [inline pl_bolts/models/detection/yolo/yolo_layers.py:85]: We recently added this to torchvision 0.8 release `torchvision.ops.generalized_box_iou`

File: pl_bolts/models/detection/yolo/yolo_layers.py
Comment (oke-aditya) [inline pl_bolts/models/detection/yolo/yolo_layers.py:126]: 1 - `box_iou` simply ?

**Review by oke-aditya — COMMENTED** [review]
The PR is great :tada:  Nice implementation of YOLO v4. It's hard to do in PyTorch since it's model is originally written in DarkNet. :sweat_smile:  Great work


A few comments, the reason is [this](https://github.com/pytorch/vision/pull/3383) pr on torchvision. Hence I think it could be safer to use `torchvision.ops` ?

Somehow torchvision dependency continues here as well I proposed a few solutions in #554 .
We actually nativelely implemented ops in bolts #347 #370 . But either we need to update them cause of above issue or use torchvision implementation.

IIRC torchvision 0.7 did not document these functions for IoU, box_area, and they were not stable.
GIoU and Tests were added in 0.8 to make it generally available.

File: pl_bolts/models/detection/yolo/yolo_layers.py
Comment (senarvi) [reply pl_bolts/models/detection/yolo/yolo_layers.py:73]: @oke-aditya It's basically the same thing, but performs the operation elementwise on two sets of N boxes, producing N results. The torchvision op performs the operation on every pair formed from two sets of boxes, producing an N*M matrix. In other words, this is the same as taking the diagonal of `torchvision.ops.box_iou`, however, that would take quadratic time. The most generic and consistent way would probably be that the torchvision `box_iou` and `generalized_box_iou` functions would perform the elementwise operation, and for creating a pairwise matrix one would add empty dimensions to the inputs (or include an optional argument in those functions to perform the elementwise operation).

File: pl_bolts/models/detection/yolo/yolo_layers.py
Comment (senarvi) [reply pl_bolts/models/detection/yolo/yolo_layers.py:100]: See the discussion about `_elementwise_iou` above.

File: pl_bolts/models/detection/yolo/yolo_layers.py
Comment (senarvi) [reply pl_bolts/models/detection/yolo/yolo_layers.py:141]: To be as generic as possible, any loss function can be given to the detection layer as a parameter. So something like this would be possible: `lambda inputs, target: 1.0 - _elementwise_iou(inputs, target)`. Not that `lambda inputs, target: 1.0 - box_iou(inputs, target)` is a different thing. It calculates the loss between every (detector, target) pair, while YOLO matches the detectors to the targets and only calculates the loss from the matched pairs only.

**Comment by senarvi** [comment]
> The PR is great 🎉 Nice implementation of YOLO v4. It's hard to do in PyTorch since it's model is originally written in DarkNet. 😅 Great work
> 
> A few comments, the reason is [this](https://github.com/pytorch/vision/pull/3383) pr on torchvision. Hence I think it could be safer to use `torchvision.ops` ?

Thanks! Indeed it's somewhat problematic how this project overlaps with torchvision. I definitely agree that I would rather use torchvision functions than copy them here, for the exact reason that you pointed out. We could use `box_iou(images, target).diagonal()` if we think it's worth the performance hit.

File: pl_bolts/models/detection/yolo/__init__.py
Comment (akihironitta) [inline pl_bolts/models/detection/yolo/__init__.py:4]: Could we make this file empty? Otherwise, we'll have two ways to import these names:
```python
from pl_bolts.models.detection import Yolo       # <- This is fine.
from pl_bolts.models.detection.yolo import Yolo  # <- We don't want this to keep the API simple. (I think)
```

**Review by akihironitta — COMMENTED** [review]
@senarvi This is well documented and a great addition to Bolts! I just left some comments, so would you mind having a look?

File: pl_bolts/models/detection/yolo/yolo_layers.py
Comment (akihironitta) [reply pl_bolts/models/detection/yolo/yolo_layers.py:141]: Same as https://github.com/PyTorchLightning/pytorch-lightning-bolts/pull/552#discussion_r575473956, right?

File: pl_bolts/models/detection/yolo/yolo_layers.py
Comment (akihironitta) [reply pl_bolts/models/detection/yolo/yolo_layers.py:73]: So, here we have two options (right?):
1. to use `_elementwise_iou` as is, or
2. to remove `_elementwise_iou` and use `torchvision.ops.box_iou` (but increase computational cost from `O(N)` to `O(N^2)`)

@oke-aditya @PyTorchLightning/bolts-contributors @Borda Any thoughts on this? (I'm neutral on this.)

File: pl_bolts/models/detection/yolo/yolo_config.py
Comment (akihironitta) [inline pl_bolts/models/detection/yolo/yolo_config.py:14]: Single quote will be rendered in italic in the docs i.e. *get_network()*, so we need double quotes here to make it look like `get_network()`.
```suggestion
    The ``get_network()`` method returns a PyTorch module list that can be used to construct a YOLO
```

File: pl_bolts/models/detection/yolo/__init__.py
Comment (oke-aditya) [reply pl_bolts/models/detection/yolo/__init__.py:4]: I think we can just rename yolo.py `yolo_module` like we have done for other detection models

File: pl_bolts/models/detection/yolo/yolo_layers.py
Comment (oke-aditya) [reply pl_bolts/models/detection/yolo/yolo_layers.py:73]: It is fine to increase a slight computational cost. Since these are tensor operations, they are fast and batched, it will not scale up so soon.

Comment (senarvi) [reply pl_bolts/models/detection/yolo/yolo_layers.py:73]: I might add a third option, suggesting a change in the torchvision interface that supports both variants. But it's better to merge this PR first and improve it in future PRs, so I'll do like @oke-aditya suggests and use the current torchvision interface for now. At least then we'll know that it works correctly.

File: pl_bolts/models/detection/yolo/__init__.py
Comment (senarvi) [reply pl_bolts/models/detection/yolo/__init__.py:4]: @oke-aditya there's no yolo.py. There's a yolo subdirectory, like faster_rcnn.  @akihironitta FasterRCNN is also imported in both [detection/__init__.py](https://github.com/PyTorchLightning/pytorch-lightning-bolts/blob/master/pl_bolts/models/detection/__init__.py) and [detection/faster_rcnn/__init__.py](https://github.com/PyTorchLightning/pytorch-lightning-bolts/blob/master/pl_bolts/models/detection/faster_rcnn/__init__.py). Should I clear `detection/faster_rcnn/__init__.py` too, or leave them like this?

Comment (oke-aditya) [reply pl_bolts/models/detection/yolo/__init__.py:4]: Nice catch. I will solve faster RCNN in Retina Net PR. This will avoid merge conflicts too. 
Hopefully I will get retina Net completed by this week. (Really sorry for lack of time)

File: pl_bolts/models/detection/yolo/yolo_module.py
Comment (kaushikb11) [inline pl_bolts/models/detection/yolo/yolo_module.py:26]: Nit: I prefer `YOLO` in caps. @akihironitta What do you think?

File: pl_bolts/models/detection/yolo/yolo_module.py
Comment (kaushikb11) [inline pl_bolts/models/detection/yolo/yolo_module.py:172]: What not pass `optimizer: Type[Optimizer] = torch.optim.Adam` instead of `str`? Wouldn't need the conditions here.

File: pl_bolts/models/detection/yolo/yolo_module.py
Comment (akihironitta) [reply pl_bolts/models/detection/yolo/yolo_module.py:24]: I agree. @Separius Would you mind following the @kaushikb11's suggestion above?

File: pl_bolts/models/detection/yolo/yolo_layers.py
Comment (senarvi) [reply pl_bolts/models/detection/yolo/yolo_layers.py]: I'm now using `pl_bolts.utils._TORCHVISION_AVAILABLE` in a couple of places.

File: pl_bolts/models/detection/yolo/yolo_module.py
Comment (senarvi) [reply pl_bolts/models/detection/yolo/yolo_module.py]: Fixed.

File: pl_bolts/models/detection/yolo/yolo_config.py
Comment (senarvi) [reply pl_bolts/models/detection/yolo/yolo_config.py]: I replaced single quotes with double quotes in several docstrings.

File: pl_bolts/models/detection/yolo/yolo_module.py
Comment (senarvi) [reply pl_bolts/models/detection/yolo/yolo_module.py:170]: I guess I wrote it this way because it was easy to add the constructor arguments to the CLI arguments using `argparse_utils.add_argparse_args()`, but you're right - it's a lot better to give the opportunity to provide any kind of optimizer object.

File: pl_bolts/models/detection/yolo/yolo_module.py
Comment (senarvi) [reply pl_bolts/models/detection/yolo/yolo_module.py:24]: Right, the convention seems to be to write acronyms with all caps. (By the way, my handle is @senarvi.)

File: pl_bolts/models/detection/yolo/yolo_config.py
Comment (kaushikb11) [inline pl_bolts/models/detection/yolo/yolo_config.py:11]:
```suggestion
class YOLOConfiguration:
```

File: pl_bolts/models/detection/yolo/__init__.py
Comment (akihironitta) [reply pl_bolts/models/detection/yolo/__init__.py:4]: > @akihironitta FasterRCNN is also imported in both [detection/__init__.py](https://github.com/PyTorchLightning/pytorch-lightning-bolts/blob/master/pl_bolts/models/detection/__init__.py) and [detection/faster_rcnn/__init__.py](https://github.com/PyTorchLightning/pytorch-lightning-bolts/blob/master/pl_bolts/models/detection/faster_rcnn/__init__.py). Should I clear `detection/faster_rcnn/__init__.py` too, or leave them like this?

@senarvi There are similar structures in `pl_bolts/models/*` as you point out, and we are actually trying to refactor these as discussed in https://github.com/PyTorchLightning/pytorch-lightning-bolts/issues/427#issuecomment-762032327, so let's make this file empty.

File: pl_bolts/models/detection/yolo/yolo_config.py
Comment (senarvi) [reply pl_bolts/models/detection/yolo/yolo_config.py]: Fixed.

Comment (senarvi) [reply pl_bolts/models/detection/yolo/yolo_config.py]: Fixed.

File: pl_bolts/models/detection/yolo/__init__.py
Comment (senarvi) [reply pl_bolts/models/detection/yolo/__init__.py]: Fixed.

File: pl_bolts/models/detection/yolo/yolo_layers.py
Comment (senarvi) [reply pl_bolts/models/detection/yolo/yolo_layers.py]: I'm using torchvision now. By the way, the computational cost is quadratic in the _number of targets per image_, which is typically not very large. `generalized_box_iou` is not yet supported by the version of torchvision used by the PL Bolts test suite, so I added a check.

File: pl_bolts/datamodules/vocdetection_datamodule.py
Comment (senarvi) [reply pl_bolts/datamodules/vocdetection_datamodule.py:187]: Actually it doesn't. [`vocdetection_datamodule.Compose` assumes that `image_transforms` is callable](https://github.com/PyTorchLightning/pytorch-lightning-bolts/blob/3e9bddeb09a71fafc55769fa9a5c0cd7033cd330/pl_bolts/datamodules/vocdetection_datamodule.py#L30), so when normalization is used, [the default image transforms are wrapped in a `torchvision.transforms.Compose`](https://github.com/PyTorchLightning/pytorch-lightning-bolts/blob/3e9bddeb09a71fafc55769fa9a5c0cd7033cd330/pl_bolts/datamodules/vocdetection_datamodule.py#L214).

File: pl_bolts/models/detection/yolo/yolo_layers.py
Comment (senarvi) [reply pl_bolts/models/detection/yolo/yolo_layers.py]: Fixed.

File: pl_bolts/models/detection/yolo/yolo_layers.py
Comment (senarvi) [reply pl_bolts/models/detection/yolo/yolo_layers.py]: I fixed the type hints of all the constructors.

File: pl_bolts/models/detection/yolo/yolo_layers.py
Comment (akihironitta) [reply pl_bolts/models/detection/yolo/yolo_layers.py]: > number of targets per image, which is typically not very large.

Sounds reasonable! Thanks for the update!

File: pl_bolts/datamodules/vocdetection_datamodule.py
Comment (kaushikb11) [inline pl_bolts/datamodules/vocdetection_datamodule.py:157]: It's not considered good practice to have an empty list as default.

Comment (senarvi) [reply pl_bolts/datamodules/vocdetection_datamodule.py:157]: That's true. For some reason I wanted to minimize changes outside the YOLO class, but you're right. I'll change the default value to None.

File: tests/models/test_detection.py
Comment (Borda) [inline tests/models/test_detection.py:46]: can we rather have it in `tests/data` no need to be as a code

**Review by Borda — APPROVED** [review]
overall it looks good to me, mind check minor comments from @akihironitta

**Comment by kaushikb11** [comment]
@senarvi The tests failing here are not related to these PR changes. It will be fixed in #580

**Comment by senarvi** [comment]
> overall it looks good to me, mind check minor comments from @akihironitta

I think the comments from @akihironitta have been resolved. Let me know if there's still something.

File: tests/models/test_detection.py
Comment (senarvi) [reply tests/models/test_detection.py]: That's a good idea. I moved the test config to `tests/data`.

File: pl_bolts/models/detection/yolo/yolo_module.py
Comment (kaushikb11) [inline pl_bolts/models/detection/yolo/yolo_module.py:276]: @senarvi Is the `load_darnet_weights` functionality complete?

Comment (senarvi) [reply pl_bolts/models/detection/yolo/yolo_module.py:276]: It should be complete. I'm able to detect boxes using Darknet weights, and I'm able to continue training after loading Darknet weights. However, I got lower scores than when doing transfer learning using the Darknet implementation. I haven't got a chance to properly test it yet though, so I'm interested if you see any problems or if you're able to try it.

Comment (senarvi) [reply pl_bolts/models/detection/yolo/yolo_module.py:292]: I experimented with loading weights from a Darknet model. Training was very unstable, at least when using one of the truncated weight files. The Darknet weight files that are meant for transfer learning (such as [yolov4.conv.137](https://github.com/AlexeyAB/darknet/releases/download/darknet_yolo_v3_optimal/yolov4.conv.137)) are truncated up to the first detection layer, so that the number of object classes can be changed. The rest of the layers are initialized randomly. You can use them, but if you don't change the number of classes, you can load weights to all layers (such as [yolov4.weights](https://github.com/AlexeyAB/darknet/releases/download/darknet_yolo_v3_optimal/yolov4.weights)). I found that to work better. Otherwise the only way to make training stable was to use gradient clipping. It's strange because I think the Darknet implementation is not using gradient clipping. I spent some time trying to stabilize the operations such as divisions in IoU calculation, but to no avail.

**Comment by senarvi** [comment]
I changed still one detail. The boxes are directly predicted in coordinates to the input image, not in the *[0, 1]* range. There are three reasons:
* a bit fewer scaling operations,
* consistency with torchvision models, and
* better resolution for representing coordinates with half-precision floats.

**Comment by senarvi** [comment]
One more small improvement: image size from the configuration file was passed to `DetectionLayer` constructor, and any images fed to the network had to have the same size. Now the image size from the input batch is passed to its `forward()` method, so one can use a different size during inference, or - at least in theory - train with different image sizes.

**Comment by jlevy44** [comment]
Hi there! This looks like an amazing PR. What is the timeline for merging?

**Comment by kaushikb11** [comment]
The PR is good to go! But there are failing tests.
@akihironitta Could you please assist @senarvi with the failing tests, so we could go ahead with merging?

**Comment by akihironitta** [comment]
@kaushikb11 Sure, I'll look into this within 24 hours :)

**Comment by jlevy44** [comment]
Great, thanks @kaushikb11 @senarvi @akihironitta !

**Comment by senarvi** [comment]
A test was failing because `/` was used on integer tensors, which is deprecated. I fixed it, but now I don't get a proper error message anymore, why the tests are stuck. Maybe just try rerunning them? The tests were already passing earlier, so there shouldn't be anything major anymore.

**Comment by akihironitta** [comment]
@senarvi I approved and reran the tests. https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks was blocking you from running tests in CI.

File: pl_bolts/models/detection/yolo/yolo_layers.py
Comment (akihironitta) [inline pl_bolts/models/detection/yolo/yolo_layers.py:322]:
```suggestion
        image_to_grid = torch.true_divide(grid_size, image_size)
```
Could you use `torch.true_divide()` instead of `/` here, too?

**Review by akihironitta — CHANGES_REQUESTED** [review]
@senarvi Would you mind applying the suggestion below to pass the tests?

File: pl_bolts/models/detection/yolo/yolo_layers.py
Comment (senarvi) [reply pl_bolts/models/detection/yolo/yolo_layers.py]: Yes, that seems to be the problem. Fixed.

**Comment by senarvi** [comment]
Still one check fails to run. @akihironitta could you check what's wrong?

**Comment by akihironitta** [comment]
All of these tests are failing due to the unavailability of MNIST datasets (http://yann.lecun.com/exdb/mnist/index.html is 503 at this moment), so I believe the errors are related to none of the changes in this PR. Let me fix the issue in another PR.

**Comment by edgarriba** [comment]
@Borda what was the solution in the end for this mnist dataset issue ?

**Review by kaushikb11 — APPROVED** [review]

**Comment by akihironitta** [comment]
I created an issue for the MNIST problem. #632

**Review by akihironitta — APPROVED** [review]
LGTM!

The failing tests are unrelated to the changes in this PR. It's being fixed in #633.

**Comment by jlevy44** [comment]
Looks like #633 was merged. I still see a few failures. Anything I can do to help?

**Comment by akihironitta** [comment]
@senarvi Could you merge master?

**Comment by senarvi** [comment]
@akihironitta I merged master and I can see that the MNIST fix is in my [source branch](https://github.com/groke-technologies/pytorch-lightning-bolts/blob/yolo/pl_bolts/datasets/mnist_dataset.py), but the tests still fail with HTTP Error 503.

**Comment by senarvi** [comment]
Merged master again, but now test_data_monitor.py::test_base_log_interval_* tests seem to be failing.

**Comment by senarvi** [comment]
> Can there be an example of using the COCO dataset instead of the Pascal VOC dataset in the data module?

Currently there's no COCO data module. Adding a COCO data module is trivial, but one of the principles of this project is to keep pull requests simple and not to include several things in one pull request. So I don't know. Should I include a COCO data module in this pull request?

**Comment by hammacktony** [comment]
> > Can there be an example of using the COCO dataset instead of the Pascal VOC dataset in the data module?
> 
> Currently there's no COCO data module. Adding a COCO data module is trivial, but one of the principles of this project is to keep pull requests simple and not to include several things in one pull request. So I don't know. Should I include a COCO data module in this pull request?

It might be beneficial to remove the data module part - since it is so tied to VOC, and have another PR or set of PRs that implement the different types of datamodules.

I tried to use the VOC one as a template for COCO, and I just couldn't get where I needed to. That's why I was asking.

**Comment by senarvi** [comment]
> > > Can there be an example of using the COCO dataset instead of the Pascal VOC dataset in the data module?
> > 
> > 
> > Currently there's no COCO data module. Adding a COCO data module is trivial, but one of the principles of this project is to keep pull requests simple and not to include several things in one pull request. So I don't know. Should I include a COCO data module in this pull request?
> 
> It might be beneficial to remove the data module part - since it is so tied to VOC, and have another PR or set of PRs that implement the different types of datamodules.
> 
> I tried to use the VOC one as a template for COCO, and I just couldn't get where I needed to. That's why I was asking.

What exactly do you suggest removing? There's a small change to the existing VOC data module that allows passing transforms for both images and target boxes, for example, resizing both the image and its target boxes. This lets us see that training on VOC data works. The dummy CLI is similar to that of the currently only object detection model, FasterRCNN, and is meant more like sample code rather than a useful tool. I agree that a pull request for COCO data module would be useful, and could also update the CLIs so that they support COCO data.

Implementing a data module for COCO should be quite easy using the VOC data module as a template. Just remember to convert the coordinates from (x1, y1, width, height) to (x1, y1, x2, y2).

**Comment by Borda** [comment]
@senarvi could you check the failing tests? and mind allow edit for the maintainers... :rabbit:

**Comment by senarvi** [comment]
> @senarvi could you check the failing tests? and mind allow edit for the maintainers... 🐰

@Borda the test was failing because I was introduced a bug in the MSE loss, when I fixed the IoU loss. Both should work now. How do I allow edit for the maintainers?

**Comment by senarvi** [comment]
> > @senarvi could you check the failing tests? and mind allow edit for the maintainers... 🐰
> 
> @Borda the test was failing because I introduced a bug in the MSE loss, when I fixed a problem in the IoU loss. Both should work now. How do I allow edit for the maintainers?

It seems that "Allow edits from maintainers" is not possible when the source repository is under an organization.

**Comment by senarvi** [comment]
@Borda I noticed also the formatting issues and fixed them by running the new formatting tools.

**Comment by senarvi** [comment]
`VOCDetectionDataModule` was updated to better conform to the new data module interface. The constructor takes batch size and the transforms take image and target. We can now pass the data module to Trainer, instead of the dataloaders. `train_dataloader()` and `val_dataloader()` still support image-only transforms, but I don't see any need for using them.

**Comment by senarvi** [comment]
The test suite found still one problem with integer division with an older PyTorch version. @Borda could you restart the CI? I don't see the error when running the tests with my PyTorch installation.

**Comment by akihironitta** [comment]
@senarvi Hi, thanks again for all your work and patience! Just restarted CI again. I'll try to be active on this PR for the next few days until we get this PR merged :]

**Comment by senarvi** [comment]
@akihironitta correct me if I'm wrong, but the remaining error was something about CIFAR dataset and not related to this pull request.

File: CHANGELOG.md
Comment (akihironitta) [inline CHANGELOG.md:91]: Can you move this line to the unreleased section?

File: pl_bolts/models/detection/__init__.py
Comment (akihironitta) [inline pl_bolts/models/detection/__init__.py:6]: nit-picking :smile: 
```suggestion
__all__ = [
    "components",
    "FasterRCNN",
    "YOLOConfiguration",
    "YOLO",
]
```

**Review by akihironitta — COMMENTED** [review]
> @akihironitta correct me if I'm wrong, but the remaining error was something about CIFAR dataset and not related to this pull request.

@senarvi You're right that all of the tests here have passed. Once the following comments gets resolved, I believe we can land this PR finally!

**Comment by senarvi** [comment]
@akihironitta ok, done.

**Pull request merged**
